### PR TITLE
Retrieve uploadable keys using new service in persistence module

### PR DESCRIPTION
**Pull request by EugenM-SAP**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

Changes to support the provisioning of pending upload keys to the DiagnosisKeyBatchAssembler when the upload job is running. There will be follow-up PRs whch deal with the filtered selection of uploadable keys, and enforcing of batch constraints.

### CODE REVIEW

File: services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    List<DiagnosisKey> fakeKeys = new ArrayList<DiagnosisKey>();
+    while (numberOfKeys >= 0) {
+      fakeKeys.add(generateRandomDiagnosisKey(consentToShare));
+      numberOfKeys--;
+    }
+    return fakeKeys;
+  }
+
+  public static DiagnosisKey generateRandomDiagnosisKey(boolean consentToShare) {
+    DiagnosisKey dummyKey = DiagnosisKey.builder().withKeyData(randomByteData())
```
Comment (fredrb) [inline services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java:43]: Return the result of this call?

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+    this.batchAssembler = batchAssembler;
+  }
 
   @Override
   public void run(ApplicationArguments args) throws Exception {
     logger.info("Running Upload Job");
+    try {
+      List<DiagnosisKey> pendingUploadKeys = uploadKeyService.getPendingUploadKeys();
+      List<DiagnosisKeyBatch> batches = batchAssembler.assembleDiagnosisKeyBatch(pendingUploadKeys);
+    } catch (Exception e) {
```
Comment (fredrb) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java:40]: None of these methods seem to throw an exception. Are you wrapping around a try/catch as a placeholder for future exceptions?

**Review by fredrb — COMMENTED** [review]
I suspect the codestyle checks are not being run. This is not related to this PR as I think it didn't run on the `feature/federation` PRs either. You can see that the some of the files you added in the persistence module don't have imports ordered alphabetically, some have spaces between them and license is missing.  

Apart from that I think the code looks good, just two comments from my side.

File: services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java (hunk 1 above)
Comment (EugenM-SAP) [reply services/upload/src/test/java/app/coronawarn/server/services/federation/upload/utils/MockData.java:43]: > I suspect the codestyle checks are not being run. This is not related to this PR as I think it didn't run on the `feature/federation` PRs either. You can see that the some of the files you added in the persistence module don't have imports ordered alphabetically, some have spaces between them and license is missing.
> 
> Apart from that I think the code looks good, just two comments from my side.

Right, currently checkstyles is not run on the federation branch.

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java (hunk 2 above)
Comment (EugenM-SAP) [reply services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/Upload.java:40]: Right, at this point the Upload logic is incomplete. Next steps is to provide the batches to the http upload logic. However in any case, any runtime exception that could be thrown in this runner should be caught, and the cron job stopped immediately. This fail-fast logic is consistent with the distribution runners.

**Review by fredrb — APPROVED** [review]

**Pull request merged**